diff --git a/release/c/wuffs-v0.3.c b/release/c/wuffs-v0.3.c
index 072fa9eb1f642b4d7023fc61bb653bcf95c540d0..3f18b721be16b005d0268fcbd4bc9f2d17a3d354 100644
--- a/release/c/wuffs-v0.3.c
+++ b/release/c/wuffs-v0.3.c
@@ -144,7 +144,7 @@ extern "C" {
 
 #elif defined(_MSC_VER)  // (#if-chain ref AVOID_CPU_ARCH_1)
 
-#if defined(_M_IX86) || defined(_M_X64)
+#if (defined(_M_IX86) || defined(_M_X64)) && !defined(_M_ARM64EC)
 #if defined(__AVX__) || defined(__clang__)
 
 // We need <intrin.h> for the __cpuid function.
